lua/entities/gmod_wire_expression2/core/npc.lua

```diff
@@ -152,3 +266,47 @@ e2function string entity:npcDisp(entity rv2)
 	if disp == 0 then return "" end
 	return DispToString(disp)
 end
+--Schedule Functions--
+
+local function strToSch(str)
+	local str2 = string.upper(str)
+	
+	return SCH[str2]
+
+end
+
+local function strToState(st)
+	local st2 = string.upper(st)
+
+	return STS[str2]
+end
+
+local function stateToStr(st)
+	local st2 = string.upper(st)
+
+	return STS[str2]
+end
+
+e2function void entity:setSchedule(string schedule)
+	if not IsValid(this) and this:IsNPC() then return end
+	if not strToSch(schedule)>=0 or not strToSch(schedule)<88 then return end
```

Comment:
This should just check for nil. Enum values are subject to change so these bounds are not reliable.

Reply:
You should also only call the strToSch function once and store its result as a local.